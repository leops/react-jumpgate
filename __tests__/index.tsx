import * as React from 'react';
import renderer from 'react-test-renderer';

import createJumpgate from '../src';

test('basic', () => {
    const { Anchor, Provider, Consumer } = createJumpgate();

    const tree = renderer.create(
        <div className="root">
            <Anchor>
                <div className="consumer">
                    <Consumer />
                </div>
                <div className="provider">
                    <Provider>
                        <div className="jump" />
                    </Provider>
                </div>
            </Anchor>
        </div>
    );

    expect(tree.toJSON()).toMatchSnapshot();

    tree.update(
        <div className="root">
            <Anchor>
                <div className="consumer">
                    <Consumer />
                </div>
                <div className="provider">
                    <Provider>
                        <div className="jump-update" />
                    </Provider>
                </div>
            </Anchor>
        </div>
    );

    expect(tree.toJSON()).toMatchSnapshot();
});

test('anchor-less <Provider />', () => {
    const { Provider } = createJumpgate();

    console.error = jest.fn();

    const render = () => {
        renderer.create(
            <Provider>
                <div className="jump" />
            </Provider>
        );
    };

    expect(render).toThrowErrorMatchingSnapshot();
});

test('anchor-less <Consumer />', () => {
    const { Consumer } = createJumpgate();

    console.error = jest.fn();

    const render = () => {
        renderer.create(<Consumer />);
    };

    expect(render).toThrowErrorMatchingSnapshot();
});
