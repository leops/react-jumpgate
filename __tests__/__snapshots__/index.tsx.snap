// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anchor-less <Consumer /> 1`] = `"Attempted to render a <Consumer /> without an <Anchor />"`;

exports[`anchor-less <Provider /> 1`] = `"Attempted to render a <Provider /> without an <Anchor />"`;

exports[`basic 1`] = `
<div
  className="root"
>
  <div
    className="consumer"
  >
    <div
      className="jump"
    />
  </div>
  <div
    className="provider"
  />
</div>
`;

exports[`basic 2`] = `
<div
  className="root"
>
  <div
    className="consumer"
  >
    <div
      className="jump-update"
    />
  </div>
  <div
    className="provider"
  />
</div>
`;
